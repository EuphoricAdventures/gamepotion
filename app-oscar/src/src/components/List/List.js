import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const styleAbstractions = {
  colours: {
    'fore': '#2e3131'
  },
  font: `
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", sans-serif;
    font-weight: 400;
    text-size-adjust: 100%;
    -webkit-font-smoothing: antialiased;
  `
}

const StyledList = styled.ul`
  .component--list {
    margin-left: 1.5rem;
    .component--list-item:first-of-type {
      margin-top: 0.25rem;
    }
  }
  .component--list + .component--list-item {
    margin-top: 0.25rem;
  }
  .component--list-item + .component--list-item {
    margin-top: 0.25rem;
  }
  .empty-text {
    padding-top: 2rem;
    padding-bottom: 2rem;
    ${styleAbstractions.font}
    text-align: center;
    color: ${styleAbstractions.colours.fore};
    opacity: 0.5;
    // background-color: navy;
  }
`

const List = ({ children, emptyText }) => {
  return (
    <StyledList className='component--list'>
      {children.length === 0 && emptyText.length > 0 &&
        <p className='empty-text'>{emptyText}</p>
      }
      {children}
    </StyledList>
  )
}

List.propTypes = {
  emptyText: PropTypes.string
}

List.defaultProps = {
  emptyText: ''
}

export default List
