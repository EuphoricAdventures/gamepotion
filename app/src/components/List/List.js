import React, { Fragment } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const StyledList = styled.ul`
  .component--list {
    margin-left: 1.5rem;
    .component--list-item:first-of-type {
      margin-top: 0.25rem;
    }
  }
  .component--list + .component--list-item {
    margin-top: 1rem;
  }
  .component--list-item + .component--list-item {
    margin-top: 0.25rem;
  }
`

const List = ({ children }) => {
  return (
    <StyledList className='component--list'>
      {children}
    </StyledList>
  )
}

List.propTypes = {
}

List.defaultProps = {
}

export default List
