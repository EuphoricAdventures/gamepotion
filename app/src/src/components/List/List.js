import React from 'react'
import styled from 'styled-components'

const StyledList = styled.ul`
  .component--list {
    margin-left: 1.5rem;
    .component--list-item:first-of-type {
      margin-top: 0.25rem;
    }
  }
  .component--list + .component--list-item {
    margin-top: 0.25rem;
  }
  .component--list-item + .component--list-item {
    margin-top: 0.25rem;
  }
`

const List = ({ children }) => {
  return (
    <StyledList className='component--list'>
      {children}
    </StyledList>
  )
}

export default List
