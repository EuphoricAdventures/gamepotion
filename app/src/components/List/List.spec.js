import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import List from './List'

it('renders', () => {
  const tree = renderer
    .create(
      <List>...</List>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
