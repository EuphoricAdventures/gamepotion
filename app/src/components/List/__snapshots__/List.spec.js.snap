// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
.c0 .component--list {
  margin-left: 1.5rem;
}

.c0 .component--list .component--list-item:first-of-type {
  margin-top: 0.25rem;
}

.c0 .component--list + .component--list-item {
  margin-top: 1rem;
}

.c0 .component--list-item + .component--list-item {
  margin-top: 0.25rem;
}

<ul
  className="component--list c0"
>
  ...
</ul>
`;
